### PR TITLE
All code for hosting and joining lobbies, as well as the server and client.

### PR DESCRIPTION
**Pull request by dietterc**
Merge the branch we worked on to our main branch.  
Contains all of the code we have worked on thus far for feature 1 and 2.

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+      lobbyCode: "",
+      lobbyPlayerList: ""   
+    };
+    this.updateUsername = this.updateUsername.bind(this);
+    this.updateLobbyCode = this.updateLobbyCode.bind(this);
+    this.onJoin = this.onJoin.bind(this);
+    this.onHost = this.onHost.bind(this);
+    
+    
+    //this.socket = socket;
```
Comment (manningj) [inline pages/index.js:32]: remove this comment

File: server/server.js [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+//list of active lobbies
+var activeGameLobbies = [];
+//assign player ID's based on an incrementing variable for now, will change to usernames later
+var nextPlayerId = 0;
+
+
+//----------objects----------
+class Lobby {
+  constructor(id) {
+    this.players = []
```
Comment (GinCS) [inline server/server.js:22]: This needs semi-colons for consistency.

File: server/server.js [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+
+
+//----------socket.io server-side----------
+io.on('connection', (socket) => {
+
+  console.log(socket.id + ' connected');
+
+  /*
+  args[0]: playerId
```
Comment (manningj) [inline server/server.js:125]: comments here? old code?

File: server/server.js [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    socket.emit("join-lobby", lobby);
+    console.log("received host-lobby signal\nActive lobbies:");
+    for(var i=0;i<activeGameLobbies.length;i++) {
+      console.log(activeGameLobbies[i].lobbyId);
+    }
+
+  })
+
+  /*
+  args[0]: playerId
```
Comment (manningj) [inline server/server.js:147]: more old code?

**Review by manningj — CHANGES_REQUESTED** [review]

File: pages/index.js (hunk 1 above)
Comment (dietterc) [reply pages/index.js:32]: >:(

File: server/server.js (hunk 3 above)
Comment (manningj) [reply server/server.js:125]: lets try and make it more clear on the next passover

**Review by ckulbaski — APPROVED** [review]
lg2m

**Review by manningj — COMMENTED** [review]
good work!

**Review by manningj — APPROVED** [review]

**Comment by manningj** [comment]
Sprint 2 code all done! Excellent work, team.

**Pull request merged**